doc/hardware/porting/board_porting.rst

```diff
@@ -617,151 +714,107 @@ The build system will print this at CMake configuration time:
-
-You may use ``0.0.0`` as a minimum revision to build for by creating the file
-:file:`plank_0_0_0.conf` in the board directory. This will be used for any
-revision lower than ``0.5.0``, for example if the user builds for
-``plank@0.1.0``.
+Similar for ``letter`` where revision ``A``, ``D``, and ``F`` could be defined
+and the user builds for ``plank@E``, the build system will target revision ``D``
+.
 
 Exact revision matching
 -----------------------
 
-Alternatively, the ``EXACT`` keyword can be given to ``board_check_revision()``
-in :file:`revision.cmake` to allow exact matches only, like this:
-
-.. code-block:: cmake
+Exact revision matching is enabled when ``exact: true`` is specified in the
+revision section in :file:`board.yml`.
 
-   board_check_revision(FORMAT MAJOR.MINOR.PATCH EXACT)
-
-With this :file:`revision.cmake`, building for ``plank@0.7.0`` in the above
-example will result in the following error message:
+When exact is defined then building for ``plank@0.7.0`` in the above example
+will result in the following error message:
 
 .. code-block:: console
 
    Board revision `0.7.0` not found.  Please specify a valid board revision.
 
-Letter revision matching
-========================
+Board revision configuration adjustment
+---------------------------------------
 
-Let's say instead that you need to support revisions ``A``, ``B``, and ``C`` of
-the ``plank`` board. Create the following additional files in the board
-directory:
+When the user builds for board ``plank@<revision>`` it is possible to make
+adjustments to the board's normal configuration.
 
-.. code-block:: none
+As described in the :ref:`default_board_configuration` and
+:ref:`board_kconfig_files` sections the board default configuration is created
+from the files :file:`<board>.dts` / :file:`<board>_<identifier>.dts` and
+:file:`<board>_defconfig` / :file:`<board>_<identifier>_defconfig`.
+When building for a specific board revision, the above files are used as
+starting point and the following board files will be used in addition:
 
-   boards/<ARCH>/plank
-   ├── plank_A.conf
-   ├── plank_A.overlay
-   ├── plank_B.conf
-   ├── plank_B.overlay
-   ├── plank_C.conf
-   ├── plank_C.overlay
-   └── revision.cmake
+- :file:`<board>_<identifier>_<revision>_defconfig`: a specific revision
+  defconfig which is only used for the board and SOC / variants identified by
+  ``<board>_<identifier>``.
 
-And add the following to :file:`revision.cmake`:
+- :file:`<board>_<revision>_defconfig`: a specific revision defconfig which is
+  used for the board regardless of the SOC / variants.
 
-.. code-block:: cmake
+- :file:`<board>_<identifier>_<revision>.overlay`: a specific revision dts
+  overlay which is only used for the board and SOC / variants identified by
+  ``<board>_<identifier>``.
 
-   board_check_revision(FORMAT LETTER)
+- :file:`<board>_<revision>.overlay`: a specific revision dts overlay which is
```

Comment:
This is for boards themselves but note that this is not true for application folders with `<board>_<identifier>.overlay`, which did work with hwmv1

Reply:
correct, revision specific overlay files must go under `<appdir>/boards/` as documented.

The `<appdir>/<board>.overlay` is from before a `<appdir>/boards/` folder was supported.
If starting to also have revision specific overlays, then it's advised to use the `<appdir>/boards` folder to minimize clutter at app toplevel.